api-controllers/ServiceFabrikApiController.js

```diff
@@ -766,13 +752,18 @@ class ServiceFabrikApiController extends FabrikBaseController {
     return Promise
       .try(() => {
         if (options.instance_id && !options.plan_id) {
-          return this.cloudController
-            .findServicePlanByInstanceId(options.instance_id)
+          return eventmesh.apiServerClient.getResource({
+              resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
+              resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR,
+              resourceId: options.instance_id
+            })
             .then(resource => {
-              options.plan_id = resource.entity.unique_id;
+              options.plan_id = _.isEmpty(_.get(resource, 'status.appliedOptions')) ? resource.spec.options.plan_id :
```

Comment:
A Todo here on why this is conditionally done would help.

Reply:
Addressed.